pkg/k8s/api.go

```diff
@@ -123,7 +123,10 @@ func NewK8sAPi(shell shell.Shell, k8sConfigFilesystemPathOverride string, apiHos
 	}
 
 	if apiHostAndPortOverride == "" {
-		apiHostAndPortOverride = config.Host
+		return &kubernetesApi{
+			apiSchemeHostAndPort: config.Host,
+			config:               config,
+		}, nil
 	}
```

Comment:
And then instead of checking `apiHostAndPortOverride` at the end of this method, I'd just return:

```
		return &kubernetesApi{Config: config}, nil
```

I'd like to see one other change in this file as well. Since the override is no longer applied, then I think we can simplify the `kubernetesApi` struck with struct embedding. We can change that struct definition to:

```
type kubernetesApi struct {
	*rest.Config
}
```

And then, wherever we were previously calling `kubeapi.apiSchemeHostAndPort`, we can just call `kubeapi.Host`. Likewise where we were calling `kubeapi.config`, we can use `kubeapi.Config` instead.